hil-test/tests/spi_full_duplex_dma.rs

```diff
@@ -79,67 +79,67 @@ mod tests {
     #[test]
     #[timeout(3)]
     fn test_symmetric_dma_transfer(ctx: Context) {
-        let (tx_buffer, tx_descriptors, rx_buffer, rx_descriptors) = dma_buffers!(4);
-        let mut dma_tx_buf = DmaTxBuf::new(tx_descriptors, tx_buffer).unwrap();
+        let (rx_buffer, rx_descriptors, tx_buffer, tx_descriptors) = dma_buffers!(4);
         let dma_rx_buf = DmaRxBuf::new(rx_descriptors, rx_buffer).unwrap();
+        let mut dma_tx_buf = DmaTxBuf::new(tx_descriptors, tx_buffer).unwrap();
 
         dma_tx_buf.fill(&[0xde, 0xad, 0xbe, 0xef]);
 
         let transfer = ctx
             .spi
-            .dma_transfer(dma_tx_buf, dma_rx_buf)
+            .dma_transfer(dma_rx_buf, dma_tx_buf)
             .map_err(|e| e.0)
             .unwrap();
-        let (_, (dma_tx_buf, dma_rx_buf)) = transfer.wait();
+        let (_, (dma_rx_buf, dma_tx_buf)) = transfer.wait();
         assert_eq!(dma_tx_buf.as_slice(), dma_rx_buf.as_slice());
     }
 
     #[test]
     #[timeout(3)]
     fn test_asymmetric_dma_transfer(ctx: Context) {
-        let (tx_buffer, tx_descriptors, rx_buffer, rx_descriptors) = dma_buffers!(4, 2);
-        let mut dma_tx_buf = DmaTxBuf::new(tx_descriptors, tx_buffer).unwrap();
+        let (rx_buffer, rx_descriptors, tx_buffer, tx_descriptors) = dma_buffers!(2, 4);
```

Comment:
For some reason, [S2 HIL](https://github.com/esp-rs/esp-hal/actions/runs/10717649673/job/29718015468) is failing this test, I can also reproduce it locally.  Not sure why it only happens in S2 and when using different buffer sizes (using `dma_buffers!(4,4)` passes the test)

Reply:
@SergioGasquez I tried to run it on ESP32 and test fails there as well :/

Reply:
I recall we were having issues with that test not long ago, but perhaps I'm mistaken.

Reply:
It was disabled for s2 before https://github.com/esp-rs/esp-hal/pull/1990, but after that it was working fine (and still is,on main). Looks like something is wrong in pdma when using different size buffers, but I cant see what.

Reply:
Looks like the other tests are leaving the initial state for the test wrong. Using the following example works fine:

```rust
//! SPI Full Duplex DMA Issue Repro
//% CHIPS: esp32 esp32c2 esp32c3 esp32c6 esp32h2 esp32s2 esp32s3

#![no_std]
#![no_main]

use esp_backtrace as _;
use esp_hal::{
    dma::{Dma, DmaPriority, DmaRxBuf, DmaTxBuf},
    dma_buffers,
    gpio::Io,
    prelude::*,
    spi::{master::Spi, SpiMode},
};
use esp_println::println;

#[entry]
fn main() -> ! {
    let peripherals = esp_hal::init(esp_hal::Config::default());

    let io = Io::new(peripherals.GPIO, peripherals.IO_MUX);
    let sclk = io.pins.gpio0;
    let miso = io.pins.gpio9;
    let mosi = io.pins.gpio10;
    let cs = io.pins.gpio8;

    let dma = Dma::new(peripherals.DMA);

    let dma_channel = dma.spi2channel;

    let spi = Spi::new(peripherals.SPI2, 100.kHz(), SpiMode::Mode0)
        .with_pins(Some(sclk), Some(mosi), Some(miso), Some(cs))
        .with_dma(dma_channel.configure(false, DmaPriority::Priority0));

    let (rx_buffer, rx_descriptors, tx_buffer, tx_descriptors) = dma_buffers!(2, 4);
    let dma_rx_buf = DmaRxBuf::new(rx_descriptors, rx_buffer).unwrap();
    let mut dma_tx_buf = DmaTxBuf::new(tx_descriptors, tx_buffer).unwrap();

    dma_tx_buf.fill(&[0xde, 0xad, 0xbe, 0xef]);

    let transfer = spi
        .dma_transfer(dma_rx_buf, dma_tx_buf)
        .map_err(|e| e.0)
        .unwrap();
    let (_, (dma_rx_buf, dma_tx_buf)) = transfer.wait();
    assert_eq!(dma_tx_buf.as_slice()[0..1], dma_rx_buf.as_slice()[0..1]);
    println!("TX: {:x?}", dma_tx_buf.as_slice());
    println!("RX: {:x?}", dma_rx_buf.as_slice());
    loop {}
}
```
Also, if we comment all the tests in `spi_full_duplex_dma` but `test_asymmetric_dma_transfer` (the one that fails), the tests passes.

Reply:
That is especially sad because the MCU is supposed to be fully reset between tests.

Reply:
Just to double-check, did you try to run this example multiple times?